Src/AutoFakeItEasy/PropertyRule.cs

```diff
@@ -0,0 +1,127 @@
+        public bool IsApplicableTo(IFakeObjectCall fakeObjectCall)
+        {
+            return fakeObjectCall != null && IsProperty(fakeObjectCall.Method);
+        }
+
+        public void Apply(IInterceptedFakeObjectCall interceptedfakeObjectCall)
+        {
+            if (interceptedfakeObjectCall == null) throw new ArgumentNullException(nameof(interceptedfakeObjectCall));
+
+            var fakeObjectCall = new FakeObjectCall(interceptedfakeObjectCall);
+            var propertyCall = new PropertyCall(fakeObjectCall);
+            var methodReturnType = fakeObjectCall.Method.ReturnType;
+            if (IsSetter(fakeObjectCall))
+            {
+                this.behaviors[propertyCall] = new PropertyReturnValue(fakeObjectCall.Arguments.Last());
+            }
+            else
+            {
+                var returnValueAction = this.behaviors.GetOrAdd(
+                    propertyCall,
+                    _ => new PropertyReturnValue(this.context.Resolve(methodReturnType)));
+                returnValueAction.ApplyToCall(fakeObjectCall);
+            }
+        }
+
+        private static bool IsProperty(MethodInfo method) =>
+            method.IsSpecialName && (method.Name.StartsWith("get_", StringComparison.Ordinal) ||
+                                     method.Name.StartsWith("set_", StringComparison.Ordinal));
+
+        private static bool IsSetter(FakeObjectCall fakeObjectCall) =>
+            fakeObjectCall.Method.ReturnType == typeof(void);
+
+        private class PropertyCall
+        {
+            private readonly string methodName;
+            private readonly IList<Type> argumentTypes;
+            private readonly IList<object> arguments;
+
+            public PropertyCall(FakeObjectCall fakeCall)
+            {
+                this.methodName = fakeCall.Method.Name.Substring(4);
+                var numberOfArguments = fakeCall.Arguments.Count();
+                if (IsSetter(fakeCall))
+                {
+                    --numberOfArguments;
+                }
+
+                this.arguments = fakeCall.Arguments.Take(numberOfArguments).ToList();
+                this.argumentTypes = fakeCall.Method.GetParameters()
+                    .Take(numberOfArguments)
+                    .Select(p => p.ParameterType)
+                    .ToList();
+            }
+
+            public override bool Equals(object obj)
+            {
+                return obj is PropertyCall call &&
+                       this.methodName.Equals(call.methodName, StringComparison.Ordinal) &&
+                       this.argumentTypes.SequenceEqual(call.argumentTypes) &&
+                       this.arguments.SequenceEqual(call.arguments);
+            }
+
+            public override int GetHashCode()
+            {
+                unchecked
+                {
+                    var hashCode = -712421553;
+                    hashCode = hashCode * -1521134295 + EqualityComparer<string>.Default.GetHashCode(this.methodName);
```

Comment:
Any benefit over `this.methodName.GetHashCode()`? I thought that you expect that value might be null, but in the `Equals()` method above you don't check that...